### PR TITLE
List Ongoing Votes Feature

### LINKED ISSUE #9: List On going events
Details in the feature test:
https://github.com/AgileVentures/AsyncVoter/blob/master/features/list_ongoing_vote.feature

### PR DESCRIPTION
**Pull request by joaopapereira**
fixes #9

### CODE REVIEW

**Comment by tansaku** [comment]
interesting, so you've gone with having a size of zero represent an active vote.

I had been thinking that we'd only support a single active vote in the first instance, but I guess this works fine, given that posting on votes will be by ID so there shouldn't be any confusion about which story a vote applies to. 

So this is kind of more we need for an absolute MVP ...

Maybe I'm misinterpreting this, but have you set it up so that `GET /stories` only returns stories with size 0?  That seems a little unRESTful to me.  I feel like `GET /stories` should always return ALL stories, and if we want to modify it then that should be done by passing a parameter.  What do you think?

Also some merge conflicts here ...

**Comment by joaopapereira** [comment]
I am ok with that.
Should we go with:
GET /stories/active

**Comment by tansaku** [comment]
@joaopapereira I think ultimately `GET /stories/active` would get a story that was named `active`  - how about `GET /stories?size=0` ?

**Comment by joaopapereira** [comment]
@tansaku I think we should have a simple endpoint to get all the stories that are actively being voted instead of using the get parameters to do that

**Comment by arreche** [comment]
I like the idea about filtering actives stories.

http://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api#advanced-queries

By the way assigning `0` to `size` sounds like a little trick ;)

**Comment by joaopapereira** [comment]
What about 
GET /stories?state=active

**Comment by tansaku** [comment]
@joaopapereira I think the REST way suggests that we avoid creating custom end points and instead prefer a generic filtering mechanism - I think that will make everything much much simpler in the long run.

**Comment by tansaku** [comment]
There's interesting discussion here suggesting we use a POST:

http://stackoverflow.com/questions/5020704/how-to-design-restful-search-filtering

**Comment by tansaku** [comment]
Adding a body or params to a GET breaks caching ...

**Comment by tansaku** [comment]
Although see also http://stackoverflow.com/questions/207477/restful-url-design-for-search

After reading that I'm thinking GET /stories?size=0

**Comment by arreche** [comment]
The only thing that I don't like so much is that client should know about `GET /stories?size=0` is that it exposes implementation details.

**Comment by joaopapereira** [comment]
I agree that maybe size=0 is something that we are using for now. I would go  with my get suggestion of state=active

**Comment by tansaku** [comment]
let's work with size=0 for now - and then once we're merged we can look at state=active and see about whether we need to add that to the domain model.  I think what we want to avoid in the short term is one language for clients and a separate one for the back end - I'm working from Eric Evans concept of domain driven design where we try to keep a single consistent language throughout the system.

File: src/story/story.model.js [hunk 1]
```diff
@@ -3,8 +3,22 @@ var schema = new mongoose.Schema({
...
+{
+    timestamps: true
 });
 
+schema.statics.findBy = function(filter, callback) {
+	let filterObject = {};
+	if(filter['state'] && filter['state'] === 'active') {
+		filterObject['size'] = 0;
+	} 
+	return this.model('Story').find(filterObject).sort('name').exec(callback);
```
Comment (arreche) [inline src/story/story.model.js]: Why is it sorted by `name` instead of `date` (creation date)?

**Pull request merged**